File: recursion/factorial/factorial.ts
// Recursion: Factorial

function factorial(n: number): number {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

function main(): void {
  const n = 10;
  console.log(`${n}! = ${factorial(n)}`);
}

main();

File: recursion/factorial/factorial.java
// Recursion: Factorial
class Factorial {

    static long factorial(int n) {
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        int n = 10;
        System.out.println(n + "! = " + factorial(n));
    }
}
